### ISSUE TITLE
Please kill these short abominations one more time

### ISSUE THREAD

**Issue by Goury** [EvHaus/youtube-hide-watched#181]
Using 5.9, shorts started to appear in the search again.
Now they are mimicking non-short videos, except they are vertical and say "💩SHORTS" instead of showing a timer.

looks like
```
ytd-video-renderer.style-scope:nth-child(4) > div:nth-child(1) > ytd-thumbnail:nth-child(1) > a:nth-child(1) > div:nth-child(2) > ytd-thumbnail-overlay-time-status-renderer:nth-child(1) > div:nth-child(2) > span:nth-child(2)
```
or
```
html body ytd-app div#content.style-scope.ytd-app ytd-page-manager#page-manager.style-scope.ytd-app ytd-search.style-scope.ytd-page-manager div#container.style-scope.ytd-search ytd-two-column-search-results-renderer.style-scope.ytd-search div#primary.style-scope.ytd-two-column-search-results-renderer ytd-section-list-renderer.style-scope.ytd-two-column-search-results-renderer div#contents.style-scope.ytd-section-list-renderer ytd-item-section-renderer.style-scope.ytd-section-list-renderer div#contents.style-scope.ytd-item-section-renderer ytd-video-renderer.style-scope.ytd-item-section-renderer div#dismissible.style-scope.ytd-video-renderer ytd-thumbnail.style-scope.ytd-video-renderer a#thumbnail.yt-simple-endpoint.inline-block.style-scope.ytd-thumbnail div#overlays.style-scope.ytd-thumbnail ytd-thumbnail-overlay-time-status-renderer.style-scope.ytd-thumbnail div#time-status.style-scope.ytd-thumbnail-overlay-time-status-renderer span#text.style-scope.ytd-thumbnail-overlay-time-status-renderer
```

PS: they seem to be using the same ID "text" for each of these 💩 elements.
If there was a need for extra proof they are incompetent.

**Comment by Goury** [comment]
UPD: huge horizontally scrollable banner of 💩shorts is back too, feels like it's in the every other search

**Comment by EvHaus** [comment]
Found the issue. Give it a try in the latest v5.10 version.

**Comment by Goury** [comment]
@EvHaus thanks for supporting the script.
Gave 5.10 a try, nope, they are still leaking.

```
html.efyt-control-bar-visible.iqzdyeq.lkvaxejol body ytd-app div#content.style-scope.ytd-app ytd-page-manager#page-manager.style-scope.ytd-app ytd-search.style-scope.ytd-page-manager div#container.style-scope.ytd-search ytd-two-column-search-results-renderer.style-scope.ytd-search div#primary.style-scope.ytd-two-column-search-results-renderer ytd-section-list-renderer.style-scope.ytd-two-column-search-results-renderer div#contents.style-scope.ytd-section-list-renderer ytd-item-section-renderer.style-scope.ytd-section-list-renderer div#contents.style-scope.ytd-item-section-renderer ytd-video-renderer.style-scope.ytd-item-section-renderer div#dismissible.style-scope.ytd-video-renderer ytd-thumbnail.style-scope.ytd-video-renderer a#thumbnail.yt-simple-endpoint.inline-block.style-scope.ytd-thumbnail div#overlays.style-scope.ytd-thumbnail ytd-thumbnail-overlay-time-status-renderer.style-scope.ytd-thumbnail div#time-status.style-scope.ytd-thumbnail-overlay-time-status-renderer span#text.style-scope.ytd-thumbnail-overlay-time-status-renderer
```

```
ytd-video-renderer.style-scope:nth-child(13) > div:nth-child(2) > ytd-thumbnail:nth-child(1) > a:nth-child(1) > div:nth-child(2) > ytd-thumbnail-overlay-time-status-renderer:nth-child(1) > div:nth-child(2) > span:nth-child(2)
```

Please reopen the issue

**Comment by EvHaus** [comment]
@Goury Hmm. I can't reproduce leaks. On which page you seeing the shorts?

**Comment by Goury** [comment]
@EvHaus search results.
This crap appear if you scroll down a bit.
Horizontal crap appear and then quickly disappear.
Stealthy crap just stays.

Some times I have scroll down just a bit for it to appear.
Some times it doesn't show up until I scroll through a hundred or so normal video thumbnails.
But it does appear every time if I scroll through search results enough, no matter what I search for.

**Comment by EvHaus** [comment]
> Horizontal crap appear and then quickly disappear.

This is expected behaviour. There's a slight delay between when the content loads and when the "YouTube Hide Watched" script detects it and removes it. I'm not sure if I can make it any faster.

> Stealthy crap just stays. 

I'm not able to reproduce this. All shorts should get hidden as you scroll. Here's what I'm seeing:

https://github.com/EvHaus/youtube-hide-watched/assets/226640/c9518d9e-14db-4b0f-9bea-88bc75882420

Are you doing something different?

**Comment by Goury** [comment]
@EvHaus look closer, this crap is stealthy.
Google seems to be going out of its way to force-feed you these feces.
It appear for you at 0:19 in a form of a toddler and it doesn't go away.
And then again at 0.24 as a man with a blue dildo in his hands.
And another toddler at 0:28.
And then there's more and more of this crap.

![Снимок экрана от 2023-09-04 21-40-08](https://github.com/EvHaus/youtube-hide-watched/assets/1081221/5b006156-bdfe-435d-ac46-35784a13ad76)

**Comment by EvHaus** [comment]
Ah, I see. Published v5.11 that should hide those now.
